ddtrace/contrib/molten/patch.py

```diff
@@ -0,0 +1,159 @@
+        _u(molten.App, '__call__')
+        _u(molten.Router, 'add_route')
+
+
+def patch_app_call(wrapped, instance, args, kwargs):
+    """Patch wsgi interface for app
+    """
+    @wrapt.function_wrapper
+    def _w_start_response(wrapped, instance, args, kwargs):
+        """ Patch respond handling to set metadata """
+
+        pin = Pin.get_from(molten)
+        if not pin or not pin.enabled():
+            return wrapped(*args, **kwargs)
+
+        span = pin.tracer.current_root_span()
+        if not span:
+            return wrapped(*args, **kwargs)
+
+        status, headers, exc_info = args
+        code, _, _ = status.partition(' ')
+
+        try:
+            code = int(code)
+        except ValueError:
+            pass
+
+        span.set_tag(http.STATUS_CODE, code)
+
+        # mark 5xx spans as error
+        if 500 <= code < 600:
+            span.error = 1
+
+        return wrapped(*args, **kwargs)
+    pin = Pin.get_from(molten)
+
+    if not pin or not pin.enabled():
+        return wrapped(*args, **kwargs)
+
+    # DEV: This is safe before this is the args for a WSGI handler
+    #   https://www.python.org/dev/peps/pep-3333/
+    environ, start_response = args
+
+    # patching for extracting response code
+    start_response = _w_start_response(start_response)
+
+    request = molten.http.Request.from_environ(environ)
+    resource = func_name(wrapped)
+
+    # Configure distributed tracing
+    if config.molten.get('distributed_tracing', False):
+        propagator = HTTPPropagator()
+        # request.headers is type Iterable[Tuple[str, str]]
+        context = propagator.extract(dict(request.headers))
+        # Only need to activate the new context if something was propagated
+        if context.trace_id:
+            pin.tracer.context_provider.activate(context)
+
+    with pin.tracer.trace('molten.request', service=pin.service, resource=resource) as span:
+        span.set_tag(http.METHOD, request.method)
+        span.set_tag(http.URL, request.path)
+        span.set_tag('molten.version', molten.__version__)
+        return wrapped(environ, start_response, **kwargs)
+
+
+def patch_app_init(wrapped, instance, args, kwargs):
+    """Patch app initialization of middleware, components and renderers
+    """
+    # allow instance to be initialized before wrapping them
+    wrapped(*args, **kwargs)
+
+    # add Pin to instance
+    pin = Pin.get_from(molten)
+
+    if not pin or not pin.enabled():
+        return
+
+    # Wrappers here allow us to trace objects without altering class or instance
+    # attributes, which presents a problem when classes in molten use
+    # ``__slots__``
+
+    instance.router = WrapperRouter(instance.router)
+
+    # wrap middleware functions/callables
+    instance.middleware = [
+        WrapperMiddleware(mw)
+        for mw in instance.middleware
+    ]
+
+    # wrap components objects within injector
+    # NOTE: the app instance also contains a list of components but it does not
```

Comment:
I wonder if it is worth setting them both just for consistency?

If you think we are fine here, then not a big deal.

Reply:
I decided to remove the wrapping of the `instance.components` since it didn't have any effect on the tests or traces from the molten example. A better approach to how we patch dependency injection to deal with the `resolve` method might end up addressing this confusion.